core/services/ardupilot_manager/firmware_download/FirmwareDownload.py

```diff
@@ -232,7 +232,9 @@ def download(self, vehicle: Vehicle, platform: Platform, version: str = "") -> O
             Optional[pathlib.Path]: Temporary path for the firmware file, None if unable to download or validate file.
         """
         if platform == Platform.Navigator and vehicle == Vehicle.Sub:
-            return FirmwareDownload._download_navigator()
+            path = FirmwareDownload._download_navigator()
+            os.chmod(str(path), stat.S_IXOTH)
+            return path
```

Comment:
Oh, good one. This was introduced by my SITL PR (#289).

Line 283 should chmod the binary, as its .elf, but as it's a navigator one it returns before.

Maybe we could merge both?

Reply:
In fact, as the Navigator download is a temporary solution, I'm very ok with it.

Reply:
Check #306